rasterio/gdal.pxi

```diff
@@ -265,22 +265,70 @@ cdef extern from "gdal.h" nogil:
         GRIORA_Mode
         GRIORA_Gauss
 
-    ctypedef enum GDALColorInterp:
-        GCI_Undefined
-        GCI_GrayIndex
-        GCI_PaletteIndex
-        GCI_RedBand
-        GCI_GreenBand
-        GCI_BlueBand
-        GCI_AlphaBand
-        GCI_HueBand
-        GCI_SaturationBand
-        GCI_LightnessBand
-        GCI_CyanBand
-        GCI_YCbCr_YBand
-        GCI_YCbCr_CbBand
-        GCI_YCbCr_CrBand
-        GCI_Max
+
+IF (CTE_GDAL_MAJOR_VERSION, CTE_GDAL_MINOR_VERSION) >= (3, 10):
```

Comment:
I'm not totally sure we need that part at all since my understanding is that it is all required in rasterio owns code if it needs to use GCI_ constant directly, right?